packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      expect(updateResult1.data[updateResultSetName].id).toEqual(todo['id']);
+      expect(updateResult1.data[updateResultSetName].owner).toEqual(userName1);
+      expect(updateResult1.data[updateResultSetName].authors).toEqual([userName1, userName2]);
+      expectNullFields(updateResult1.data[updateResultSetName], ['privateContent', 'ownersContent']);
+
+      // user1 can read the allowed fields
+      const user1ReadAllowedSet = `
+        id
+        owner
+        authors
+        privateContent
+        ownerContent
+        ownersContent
+      `;
+      const getResult1 = await user1TodoHelper.get(
+        {
+          id: todo['id'],
+        },
+        user1ReadAllowedSet,
+        false,
+      );
+      checkOperationResult(getResult1, { ...todo, ...todoUpdated1, ownersContent: null }, `get${modelName}`);
+
+      const listTodosResult1 = await user1TodoHelper.list({}, user1ReadAllowedSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['id'], true);
+
+      // user2 can update the private and dynamic owners list protected fields.
+      const todoUpdated2 = {
+        id: todo['id'],
+        owner: todo['owner'],
+        authors: [userName2],
+        privateContent: 'Private Content updated 1',
+        ownersContent: 'Owners Content updated 1',
+      };
+      const user2UpdateAllowedSet = `
+        id
+        owner
+        authors
+        privateContent
+        ownersContent
+      `;
+      const updateResult2 = await user1TodoHelper.update(updateResultSetName, todoUpdated2, user2UpdateAllowedSet);
+      expect(updateResult2.data[updateResultSetName].id).toEqual(todo['id']);
+      expect(updateResult2.data[updateResultSetName].owner).toEqual(userName1);
+      expect(updateResult2.data[updateResultSetName].authors).toEqual([userName2]);
+      expectNullFields(updateResult2.data[updateResultSetName], ['privateContent', 'ownersContent']);
+
+      // user2 can read the allowed fields
+      const user2ReadAllowedSet = user2UpdateAllowedSet;
+      const getResult2 = await user2TodoHelper.get(
+        {
+          id: todo['id'],
+        },
+        user2ReadAllowedSet,
+        false,
+      );
+      checkOperationResult(getResult2, todoUpdated2, `get${modelName}`);
+
+      const listTodosResult2 = await user2TodoHelper.list({}, user2ReadAllowedSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult2, `list${modelName}`, todo['id'], true);
+
+      // unless one has delete access to all fields in the model, delete is expected to fail
+      await expect(
+        async () => await user1TodoHelper.delete(`delete${modelName}`, { id: todo['id'] }, user1CreateAllowedSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(`delete${modelName}`, 'Mutation'));
+
+      // user2 can listen to updates on the non-protected fields
```

Comment:
Do we test that user2 can also listen for updates to `ownersContent`?

Reply:
Yes in the `onUpdate` subscription since that field only allows updates.